#!/usr/bin/env python

PACKAGE = "singaboat_vrx"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("k_rho", double_t, 0, "Proportional gain for rho", 0.2, 0, 1.0)
gen.add("k_alpha", double_t, 0, "Proportional gain for alpha", 3.6, 0, 5.0)
gen.add("k_beta", double_t, 0, "Proportional gain for beta", 0.1, 0, 1.0)
gen.add("max_lin_vel", double_t, 0, "Maximum allowed linear velocity in m/s", 2.23, 0, 2.23)
gen.add("max_ang_vel", double_t, 0, "Maximum allowed angular velocity in rad/s", 0.89, 0, 0.89)

gen.add("gps_offset", double_t, 0, "GPS offset w.r.t. WAM-V along X-axis", 0.85, 0, 2.0)
gen.add("min_radius", double_t, 0, "Minimum radius for circling an animal", 2.0, 0, 5.0)
gen.add("max_radius", double_t, 0, "Maximum radius for circling an animal", 10.0, 0, 10.0)
gen.add("thresh_dist", double_t, 0, "Threshold distance to consider ASV being near an animal", 20.0, 0, 25.0)
gen.add("debug", bool_t, 0, "Flag to enable/disable debug messages", False)

exit(gen.generate(PACKAGE, "singaboat_vrx", "SemanticNavigation"))
